libs/community/langchain_community/vectorstores/bigquery_vector_search.py

```diff
@@ -0,0 +1,827 @@
+
+    To use, you need the following packages installed:
+        google-cloud-bigquery
+    """
+
+    def __init__(
+        self,
+        embedding: Embeddings,
+        project_id: str,
+        dataset_name: str,
+        table_name: str,
+        location: str = "US",
+        content_field: str = DEFAULT_CONTENT_COLUMN_NAME,
+        metadata_field: str = DEFAULT_METADATA_COLUMN_NAME,
+        text_embedding_field: str = DEFAULT_TEXT_EMBEDDING_COLUMN_NAME,
+        doc_id_field: str = DEFAULT_DOC_ID_COLUMN_NAME,
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        credentials: Optional[Any] = None,
+    ):
+        """Constructor for BigQueryVectorSearch.
+
+        Args:
+            embedding (Embeddings): Text Embedding model to use.
+            project_id (str): GCP project.
+            dataset_name (str): BigQuery dataset to store documents and embeddings.
+            table_name (str): BigQuery table name.
+            location (str, optional): BigQuery region. Defaults to
+                                      `US`(multi-region).
+            content_field (str): Specifies the column to store the content.
+                                 Defaults to `content`.
+            metadata_field (str): Specifies the column to store the metadata.
+                                  Defaults to `metadata`.
+            text_embedding_field (str): Specifies the column to store
+                                        the embeddings vector.
+                                        Defaults to `text_embedding`.
+            doc_id_field (str): Specifies the column to store the document id.
+                                Defaults to `doc_id`.
+            distance_strategy (DistanceStrategy, optional):
+                Determines the strategy employed for calculating
+                the distance between vectors in the embedding space.
+                Defaults to EUCLIDEAN_DISTANCE.
+                Available options are:
+                - COSINE: Measures the similarity between two vectors of an inner
+                    product space.
+                - EUCLIDEAN_DISTANCE: Computes the Euclidean distance between
+                    two vectors. This metric considers the geometric distance in
+                    the vector space, and might be more suitable for embeddings
+                    that rely on spatial relationships. This is the default behavior
+            credentials (Credentials, optional): Custom Google Cloud credentials
+                to use. Defaults to None.
+        """
+        try:
+            from google.cloud import bigquery
+
+            self.bq_client = bigquery.Client(
+                project=project_id, location=location, credentials=credentials
+            )
+        except ModuleNotFoundError:
+            raise ImportError(
+                "Please, install or upgrade the google-cloud-bigquery library: "
+                "pip install google-cloud-bigquery"
```

Comment:
should we pin to some minimal version of the SDK, or it doesn't matter?

Reply:
ASAIK, it doesn't require the version. https://pypi.org/project/google-cloud-bigquery/